File: Swift/ExampleTests/TypesTests.swift
import XCTest

final class TypesTests: XCTestCase {

    func testDataTypeDollar() {
        let glue = TypesGlue()
        glue.examplesDataTypeDollar([
            ValidValuesString(fromArray: ["0", "true", ""]),
            ValidValuesString(fromArray: ["0.01", "true", ""]),
            ValidValuesString(fromArray: ["-1", "false", "Negative not allowed"]),
            ValidValuesString(fromArray: ["0.001", "false", "Only 2 decimal digits"]),
        ])
    }

    func testDataTypeSimpleText() {
        let glue = TypesGlue()
        glue.examplesDataTypeSimpleText([
            ValidValuesString(fromArray: ["abc", "y", ""]),
            ValidValuesString(fromArray: ["ab.", "y", "period okay"]),
            ValidValuesString(fromArray: ["1234567890", "y", "digits"]),
            ValidValuesString(fromArray: ["@", "n", ""]),
            ValidValuesString(fromArray: ["-a-b", "y", "hyphens"]),
        ])
    }

}

File: CSharp/SpecStudioExamplesTestsCSharp/Types_Tests.cs
namespace SpecStudioExamplesTestsCSharp.Types{
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Collections.Generic;
using SpecStudioExamplesTestsCSharp.common;
using production;

[TestClass]
public class Types{

// -------------------------
// DataType Tests
// -------------------------
[TestMethod]
public void DataType_Dollar(){
     Types_glue glue = new Types_glue();
     List<ValidValuesString> objectList1 = new List<ValidValuesString>{
         new ValidValuesString("0","true",""),
         new ValidValuesString("0.01","true",""),
         new ValidValuesString("-1","false","Negative not allowed"),
         new ValidValuesString("0.001","false","Only 2 decimal digits"),
     };
     glue.Examples_DataType_Dollar(objectList1);
}

[TestMethod]
public void DataType_SimpleText(){
     Types_glue glue = new Types_glue();
     List<ValidValuesString> objectList2 = new List<ValidValuesString>{
         new ValidValuesString("abc","y",""),
         new ValidValuesString("ab.","y","period okay"),
         new ValidValuesString("1234567890","y","digits"),
         new ValidValuesString("@","n",""),
         new ValidValuesString("-a-b","y","hyphens"),
     };
     glue.Examples_DataType_SimpleText(objectList2);
}

}
}
